### DISCUSSION TITLE
using fprime-util build to build RPI demo

### DISCUSSION THREAD

**Discussion by popadirka** [nasa/fprime#1092] [Q&A]
Hi ,
I am trying to find out what I did wrong. 
I pulled out my development environment which include RPI ?( Raspberry Pi demo )
I try to build it just out of box , I did fprime-util purge ., fprime-util generate, and fprime-util build failed  building target FW_Port
I can see message Generating MsmToTlmPortAc.cpp and    MsmToTlmPortAc.hpp and I can see message 
No such file or directory.
App/Ports/Cmakefiles/App_Ports.dir/build.make:234 recipe for target  App/Ports/MsmToTlmPortAc.cpp failed.
What Can cause and how I can fix this build error. For some reason MsmToTlmPortAc.cpp was not generate , but is only my speculation.
Thanks in advance
\

**Comment by LeStarch** [comment]
It looks like `MsmToTlmPort` is a custom port type you created.  Does the `MsmToTlmPortAi.xml` exist in the right place?  Were there errors when that was read and used to autocode?

**Reply by sheershak-agarwal** [reply]
This problem was solved by reinstalling fprime directory. I had fprime repo embedded in my repo(as a subfolder and not as git submodule) but for some reason it threw errors while compiling. I changed cmake files and some driver files so I didn't want to make those changes again and again(which is why I embed that). @LeStarch Do you know why it was necessary to reinstall the fprime?

**Reply by LeStarch** [reply]
It sounds like something got corrupted, or changed.  Did you try a `fprime-util purge; fprime-util generate; fprime-util build`  to clean completely and rerun?  This would happen if you fully replaced fprime with a new copy....so you may be able to avoid a full replace the in future by purging the build cache with those above command.

**Reply by sheershak-agarwal** [reply]
Yes I did try purging. But it only cleans up the auto-generated folders like build-fprime-automatic and build-artifacts. It has nothing to do with whats inside the fprime folder.
